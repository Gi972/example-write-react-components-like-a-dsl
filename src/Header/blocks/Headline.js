import React from 'react'

import './Headline.css'

export default function Headline({
  children = "Welcome to React"
}) {
  return(
    <h1 className="App-title">{children}</h1>
  )
}
