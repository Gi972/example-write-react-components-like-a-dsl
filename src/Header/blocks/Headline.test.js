import React from 'react'

import Headline from './Headline'

describe('<Headline></Headline>', () => {
  it('renders', () => {
    expect(<Headline/>).toMatchSnapshot()
  })
  it('renders overwriting text', () => {
    expect(<Headline>Other Headline</Headline>).toMatchSnapshot()
  })
})
