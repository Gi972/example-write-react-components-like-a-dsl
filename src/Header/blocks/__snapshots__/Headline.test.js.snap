// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Headline></Headline> renders 1`] = `<Headline />`;

exports[`<Headline></Headline> renders overwriting text 1`] = `
<Headline>
  Other Headline
</Headline>
`;
